vg/vggio/context.go

```diff
@@ -0,0 +1,90 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package vggio // import "gonum.org/v1/plot/vg/vggio"
+
+import (
+	"image"
+	"image/color"
+
+	"gioui.org/f32"
+	"gioui.org/op"
+	"gioui.org/op/clip"
+
+	"gonum.org/v1/plot/vg"
+)
+
+// context holds state about the Gio backing store.
+// context provides methods to translate between Gio values (reference frame,
+// operations and stack) and their plot/vg counterparts.
+type context struct {
+	color   color.Color // color is the current color.
+	linew   vg.Length   // linew is the current line width.
+	pattern []vg.Length // pattern is the current line style.
+	offset  vg.Length   // offset is the current line style.
+	stk     op.StackOp  // stk is the Gio op context stack.
+}
+
+// ctxops holds a stack of Gio operations.
+type ctxops struct {
+	ops *op.Ops   // ops is the Gio operations vggio is drawing on.
+	ctx []context // ctx is the stack of Gio operations vggio is manipulating.
+
+	w   vg.Length // w is the canvas window width.
+	h   vg.Length // h is the canvas window height.
+	dpi float64   // dpi is the canvas window dots per inch resolution.
+}
```

Comment:
Can you move this above `context`? This refers to that, so this should come first.